src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -187,7 +191,7 @@ class QueueIO[T <: Data](private val gen: T, val entries: Int) extends Bundle
   * @param flow True if the inputs can be consumed on the same cycle (the inputs "flow" through the queue immediately).
   * The ''valid'' signals are coupled.
   * @param useSyncReadMem True uses SyncReadMem instead of Mem as an internal memory element.
-  *
+  * @param hasFlush True if generated queue requires a flush feature
```

Comment:
I think we should leave a comment here to indicate `flush` signal being couple to `ready` signal.
This comment can help user to debug the combinational cycle issue introduced by this signal.

Reply:
Agree with @aswaterman's comment, it will be much better if we don't add additional arc from `flush` to `ready`.

I still suggest using my previous review in https://github.com/chipsalliance/chisel3/pull/2030#discussion_r668208072

Reply:
this is no longer the case, i think we can resolve this comment